src/AlgoliaSearchCore.js

```diff
@@ -154,7 +154,9 @@ AlgoliaSearchCore.prototype.setExtraHeader = function(name, value) {
 * @param algoliaAgent the agent to add
 */
 AlgoliaSearchCore.prototype.addAlgoliaAgent = function(algoliaAgent) {
-  this._ua += ';' + algoliaAgent;
+  if (this._ua.indexOf(';' + algoliaAgent) === -1) {
```

Comment:
What would be the pros of that approach? Because that was my initial idea but I prefer the indexOf approach because we don't need to hold any reference, we just check inside the existing string.